File: reactive-prog/src/main/java/br/com/adriane/demo/reactiveprog/controller/ContactController.java
package br.com.adriane.demo.reactiveprog.controller;

import br.com.adriane.demo.reactiveprog.amqp.ContactMessage;
import br.com.adriane.demo.reactiveprog.amqp.Producer;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.core.AmqpTemplate;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequestMapping("/contact")
public class ContactController {

    private final Producer producer;

    public ContactController(final Producer producer) {
        this.producer = producer;
    }

    @GetMapping
    public void sendMessage() {
        ContactMessage contactMessage = new ContactMessage();
        contactMessage.setId(1);
        contactMessage.setName("Adriane Rodrigues");
        contactMessage.setEmail("adriane@email.com.br");

        log.info("Sending... {}", contactMessage);
        producer.send(contactMessage);
    }
}

File: ddd-demo/src/main/kotlin/br/com/adriane/demo/ddddemo/web/controllers/ContactController.kt
package br.com.adriane.demo.ddddemo.web.controllers

import br.com.adriane.demo.ddddemo.domain.AgendaRepository
import br.com.adriane.demo.ddddemo.web.models.ContactResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/contacts")
class ContactController(
    private val agendaRepository: AgendaRepository
) {

    @GetMapping("/{contactId}")
    fun findContactFromId(@PathVariable contactId: Int) : ContactResponse? {
        return agendaRepository.findContact(contactId)
    }
}